### PR TITLE
Solutions to The Algorithms

### PR DESCRIPTION
**Pull request by Thommzy**
algorithms solved. Kindly review

### CODE REVIEW

File: src/binary-reversal/index.js [hunk 1]
```diff
@@ -3,6 +3,47 @@
...
+function binaryReversal(value) {
+  let stringToNumber = parseInt(value);
+  let decimalToBinary = stringToNumber.toString(2);
+
+  if (decimalToBinary.length < 8) {
+    let paddedBinary = "0000000" + stringToNumber.toString(2);
+    let actualBinary = paddedBinary.substring(
+      paddedBinary.length - 8,
+      paddedBinary.length
+    );
```
Comment (bondz) [inline src/binary-reversal/index.js:15]: Interesting. Look into the `padStart` method for string.

File: src/binary-reversal/index.js [hunk 2]
```diff
@@ -3,6 +3,47 @@
...
+
+    let splitted = actualBinary.split("");
+    let reversedSplittedString = splitted.reverse();
+    let joinedReversedSplittedString = reversedSplittedString.join("");
+
+    let binaryLength = joinedReversedSplittedString.length;
+    var sum = 0;
```
Comment (bondz) [inline src/binary-reversal/index.js:22]: Avoid using `var`.

File: src/binary-reversal/index.js [hunk 3]
```diff
@@ -3,6 +3,47 @@
...
+
+    for (let i = 0, fac = 1; i < binaryLength; i++, fac *= 2)
+      sum += fac * joinedReversedSplittedString[binaryLength - i - 1];
```
Comment (bondz) [inline src/binary-reversal/index.js:25]: Embrace braces. Look into `parseInt`

File: src/binary-reversal/index.js [hunk 4]
```diff
@@ -3,6 +3,47 @@
...
+
+    return sum.toString();
+  }
+
+  let nonPaddedBinary = stringToNumber.toString(2);
```
Comment (bondz) [inline src/binary-reversal/index.js:30]: I think the entire block below could've been avoided and you could've done a single pass through the string.

File: src/list-sorting/index.js [hunk 5]
```diff
@@ -1,3 +1,15 @@
-function listSorting(needle, haystack) {}
+function listSorting(needle, haystack) {
+  let haystackLength = haystack.length;
+
+  for (let j = haystackLength - 1; j >= 0; j--) {
+    let index = haystack[j];
+
+    if (typeof index != "object") return haystack.lastIndexOf(needle);
```
Comment (bondz) [inline src/list-sorting/index.js:7]: Embrace braces. Prefer `Array.isArray` when checking if an object is an array.

File: src/list-sorting/index.js [hunk 6]
```diff
@@ -1,3 +1,15 @@
-function listSorting(needle, haystack) {}
+function listSorting(needle, haystack) {
+  let haystackLength = haystack.length;
+
+  for (let j = haystackLength - 1; j >= 0; j--) {
```
Comment (bondz) [inline src/list-sorting/index.js:4]: Think you can find a more descriptive variable name than `j`

File: src/binary-reversal/index.js (hunk 1 above)
Comment (Thommzy) [reply src/binary-reversal/index.js:15]: > Interesting. Look into the `padStart` method for string.

okay... thanks

File: src/binary-reversal/index.js (hunk 2 above)
Comment (Thommzy) [reply src/binary-reversal/index.js:22]: > Avoid using `var`.

Thanks.. I will

File: src/binary-reversal/index.js (hunk 3 above)
Comment (Thommzy) [reply src/binary-reversal/index.js:25]: > Embrace braces. Look into `parseInt`

Okay... Thanks

File: src/list-sorting/index.js (hunk 5 above)
Comment (Thommzy) [reply src/list-sorting/index.js:7]: > Embrace braces. Prefer `Array.isArray` when checking if an object is an array.

Thanks... will look into that.

File: src/list-sorting/index.js (hunk 6 above)
Comment (Thommzy) [reply src/list-sorting/index.js:4]: > Think you can find a more descriptive variable name than `j`

Thanks... i will next time

**Pull request merged**